kartothek/core/index.py

```diff
@@ -688,9 +689,27 @@ def store(self, store: KeyValueStore, dataset_uuid: str) -> str:
                 timestamp=quote(self.creation_time.isoformat()),
             )
 
-        table = _index_dct_to_table(self.index_dct, self.column, self.dtype)
+        # The arrow representation of index_dct requires a large amount of memory because strings are duplidated and
+        # flattened into the buffer. To avoid a high peak memory usage, split the index_dct into chunks and only convert
+        # one chunk a time to arrow.
+        parts_iter = partition_all(10_000, self.index_dct.items())
+
+        # Get first table explicit because its schema is required for ParquetWriter.
+        try:
+            table = _index_dct_to_table(dict(next(parts_iter)), self.column, self.dtype)
+        except StopIteration:
+            # index_dct was empty, just pass it entirely
```

Comment:
why?

Reply:
index dict empty -> index_dct.items() creates an empty iterator -> next causes StopIteration exception

We still need to call the function to get the arrow table. we do store a parquet file after all, and I wouldn't want to change that.